php-checks/src/main/java/org/sonar/php/checks/WeakSSLProtocolCheck.java

```diff
@@ -0,0 +1,161 @@
+    STREAM_CONTEXT_CREATE, Arrays.asList(
+      "STREAM_CRYPTO_METHOD_ANY_CLIENT",
+      "STREAM_CRYPTO_METHOD_ANY_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_SERVER"),
+    STREAM_SOCKET_ENABLE_CRYPTO, Arrays.asList(
+      "STREAM_CRYPTO_METHOD_SSLv2_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv3_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv23_CLIENT",
+      "STREAM_CRYPTO_METHOD_ANY_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLS_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_CLIENT",
+      "STREAM_CRYPTO_METHOD_SSLv2_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv3_SERVER",
+      "STREAM_CRYPTO_METHOD_SSLv23_SERVER",
+      "STREAM_CRYPTO_METHOD_ANY_SERVER",
+      "STREAM_CRYPTO_METHOD_TLS_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_0_SERVER",
+      "STREAM_CRYPTO_METHOD_TLSv1_1_SERVER"
+    )
+  );
+
+  private static final String MESSAGE = "Change this code to use a stronger protocol";
+  private AssignmentExpressionVisitor assignmentExpressionVisitor;
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    assignmentExpressionVisitor = new AssignmentExpressionVisitor(context().symbolTable());
+    tree.accept(assignmentExpressionVisitor);
+    super.visitCompilationUnit(tree);
+  }
+
+  @Override
+  public void visitFunctionCall(FunctionCallTree tree) {
+    String functionName = CheckUtils.getFunctionName(tree);
+    if (STREAM_CONTEXT_CREATE.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (!arguments.isEmpty()) {
+        checkSSLConfig(arguments.get(0));
+      }
+    }
+    if (STREAM_SOCKET_ENABLE_CRYPTO.equals(functionName)) {
+      SeparatedList<ExpressionTree> arguments = tree.arguments();
+      if (arguments.size() > 2) {
+        checkWeakProtocol(getAssignedValue(arguments.get(2)), STREAM_SOCKET_ENABLE_CRYPTO);
+      }
+    }
+    super.visitFunctionCall(tree);
+  }
+
+  private void checkSSLConfig(ExpressionTree expressionTree) {
+    ExpressionTree param = getAssignedValue(expressionTree);
+    if (!param.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+      return;
+    }
+    ArrayInitializerTree config = (ArrayInitializerTree) param;
+    getProperty(config, "SSL")
+      .flatMap(sslConfig -> {
+        if (sslConfig.is(Tree.Kind.ARRAY_INITIALIZER_BRACKET, Tree.Kind.ARRAY_INITIALIZER_FUNCTION)) {
+          return getProperty((ArrayInitializerTree) sslConfig, "crypto_method");
+        }
+        return Optional.empty();
+      })
+      .ifPresent(value -> checkWeakProtocol(value, STREAM_CONTEXT_CREATE));
+  }
+
+  private void checkWeakProtocol(ExpressionTree expressionTree, String functionName) {
+    Stream<ExpressionTree> expressionTrees = expressionTree.is(Tree.Kind.BITWISE_OR)
```

Comment:
nitpick: rename `expressionTrees` to `protocols` (and `value` below to `protocol`)